app/services/credit_notes/void_service.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module CreditNotes
+  class VoidService < BaseService
+    def initialize(credit_note:)
+      @credit_note = credit_note
+
+      super
+    end
+
+    def call
+      return result.not_found_failure!(resource: 'credit_note') if credit_note.nil?
+
+      result.credit_note = credit_note
+      return result.not_allowed_failure!(code: 'no_voidable_amount') unless credit_note.voidable?
+
+      credit_note.update!(
```

Comment:
Just a suggestion.

Maybe create a `mark_as_void` or similar method in the credit note model instead ?

Reply:
Good catch!